ldms/src/ldmsd/ldmsd_config.c

```diff
@@ -1005,29 +1001,20 @@ int listen_on_ldms_xprt(ldmsd_listen_t listen)
 				    .ai_flags = AI_PASSIVE };
 	char port_buff[8];
 
-	listen->x = ldms_xprt_new_with_auth(listen->xprt, ldmsd_linfo,
-			listen->auth_name, listen->auth_attrs);
 	if (!listen->x) {
-		char *args = av_to_string(listen->auth_attrs, AV_EXPAND);
-		ldmsd_log(LDMSD_LERROR,
-			  "'%s' transport creation with auth '%s' "
-			  "failed, error: %s(%d). args='%s'. Please check transport "
-			  "configuration, authentication configuration, "
-			  "ZAP_LIBPATH (env var), and LD_LIBRARY_PATH.\n",
-			  listen->xprt,
-			  auth_name,
-			  ovis_errno_abbvr(errno),
-			  errno, args ? args : "(empty conf=)");
-		free(args);
-		cleanup(6, "error creating transport");
+		ldmsd_log(LDMSD_LCRITICAL, "The listening endpoint '%s:%d' "
```

Comment:
I think this error message is misleading. What does 'before LDMSD listens on it mean?' Doesn't the change in the error message remove any clues to the administrator as to why the transport creation might have failed.

Reply:
With the change, `listen->x`  must be created before `listen_on_ldms_xprt` is called. The condition `if (!listen->x)` verifies that `listen->x` has really been created beforehand instead of verifying whether the call to `ldms_xprt_new_with_auth()` succeeds or not.

The change removes the creation of `listen->x` in the `listen_on_ldms_xprt()` function because without this change `listen->x` is already created in `main()` when it processes the -x option. Since the function name is `listen_on_ldms_xprt()`, I made the function only listens on the LDMS transport.